File: app/src/main/java/tr/name/fatihdogan/books/api/response/VolumeInfo.java
package tr.name.fatihdogan.books.api.response;

import android.support.annotation.Keep;
import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Keep
@SuppressWarnings({"WeakerAccess", "unused"})
public class VolumeInfo {

    //region PrintType
    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
            PRINT_TYPE_BOOK,
            PRINT_TYPE_MAGAZINE

    })
    @interface PrintType {

    }

    public static final String PRINT_TYPE_BOOK = "BOOK";
    public static final String PRINT_TYPE_MAGAZINE = "MAGAZINE";
    //endregion

    public String title;
    public String subtitle;
    public String[] authors;
    public String publisher;
    public String publishedDate;
    public String description;
    public IndustryIdentifiers[] industryIdentifierses;
    public ReadingModes readingModes;
    public Integer pageCount;
    public Dimensions dimensions;
    @PrintType
    public String printType;
    public String mainCategory;
    public String[] categories;
    public Double averageRating;
    public Integer ratingsCount;
    public String maturityRating;
    public Boolean allowAnonLogging;
    public PanelizationSummary panelizationSummary;
    public String contentVersion;
    public ImageLinks imageLinks;
    public String language;
    public String previewLink;
    public LayerInfo layerInfo;
    public String infoLink;
    public String canonicalVolumeLink;
}

File: app/src/main/java/com/fthdgn/books/api/response/VolumeInfo.kt
@file:Suppress("unused")

package com.fthdgn.books.api.response

import android.support.annotation.Keep
import android.support.annotation.StringDef

@Keep
class VolumeInfo {

    var title: String? = null
    var subtitle: String? = null
    var authors: Array<String>? = null
    var publisher: String? = null
    var publishedDate: String? = null
    var description: String? = null
    var industryIdentifierses: Array<IndustryIdentifiers>? = null
    var readingModes: ReadingModes? = null
    var pageCount: Int? = null
    var dimensions: Dimensions? = null
    @PrintType
    var printType: String? = null
    var mainCategory: String? = null
    var categories: Array<String>? = null
    var averageRating: Double? = null
    var ratingsCount: Int? = null
    var maturityRating: String? = null
    var allowAnonLogging: Boolean? = null
    var panelizationSummary: PanelizationSummary? = null
    var contentVersion: String? = null
    var imageLinks: ImageLinks? = null
    var language: String? = null
    var previewLink: String? = null
    var layerInfo: LayerInfo? = null
    var infoLink: String? = null
    var canonicalVolumeLink: String? = null

    companion object {
        @Retention(AnnotationRetention.SOURCE)
        @StringDef(PRINT_TYPE_BOOK, PRINT_TYPE_MAGAZINE)
        internal annotation class PrintType

        const val PRINT_TYPE_BOOK = "BOOK"
        const val PRINT_TYPE_MAGAZINE = "MAGAZINE"
    }
}
